src/material/autocomplete/autocomplete.ts

```diff
@@ -336,4 +343,37 @@ export class MatAutocomplete extends _MatAutocompleteBase {
   @ContentChildren(MatOption, {descendants: true}) options: QueryList<MatOption>;
   protected _visibleClass = 'mat-mdc-autocomplete-visible';
   protected _hiddenClass = 'mat-mdc-autocomplete-hidden';
+
+  constructor(
+    _changeDetectorRef: ChangeDetectorRef,
+    _elementRef: ElementRef<HTMLElement>,
+    @Inject(MAT_AUTOCOMPLETE_DEFAULT_OPTIONS) defaults: MatAutocompleteDefaultOptions,
+    platform?: Platform,
+  ) {
+    super(_changeDetectorRef, _elementRef, defaults, platform);
+
+    this._hideSingleSelectionIndicator = defaults.hideSingleSelectionIndicator ?? false;
```

Comment:
Ok. This PR doesn't add a new usage of DI. Deprecating a constructor parameter would be a breaking change, which would go in a major release. What should we do in the meantime?

I can make the defaults protected, which is how some of our other components do it.

```
  constructor(
    _changeDetectorRef: ChangeDetectorRef,
    _elementRef: ElementRef<HTMLElement>,
    @Inject(MAT_AUTOCOMPLETE_DEFAULT_OPTIONS) protected _defaults: MatAutocompleteDefaultOptions,
    platform?: Platform,
  ) {
```

Reply:
Which parameter was deprecated? As far as I can tell, the constructor signature is the same as before.

Reply:
AFAIK, switching to using `inject` would require deprecating the `defaults` parameter.

As the PR is currently implemented, nothing is deprecated. This PR implements the constructor of derived class to take same arguments as base classes. That's so the derived class can have access to the `MAT_AUTOCOMPLETE_DEFAULT_OPTIONS`.

The above snippet is a suggested alternative to implementing the constructor of the derived class. It adds the `protected` keyword so that the derived class can access the `MAT_AUTOCOMPLETE_DEFAULT_OPTIONS`.

Reply:
I think it's fine to just make it protected in the base class.

Reply:
Thanks, I changed it to make the defaults protected in the base class, which means that we don't need to touch the constructor of the derived class anymore 👍